cpacker/MemGPT
# refactor!: Migrate users + agent information into storage connectors

**Please describe the purpose of this pull request.**

- Refactor `Config` to `User` record
- Refactor `AgentConfig` to `AgentState` record
- Refactor `Agent` init and save to use `AgentState`
 
---

Todos 
- [x] Add `user_id` field to commands (default to `config.anon_clientid` if not provided)
  - [x] Data loading (`cli_load.py`) 
  - [x] Configuration (`cli_configure.py`) 
    - [x] Create row in `User` table if `config.anon_clientid` does not exist 
  - [x] MemGPT `run` (`cli.py`) 
    - [x] Lookup agents from metadata store, and insert new `AgentState` objects if new one is created 
  - [x] Metadata listing (e.g. listing sources, agents, etc.) 
- [x] Migrate all references to `StorageConnector` to pass in `user_id`, and optionally `agent_id` (if recall/archival memory)
- [x] Check that `run` and `configure` work
- [x] Ensure tests are all working 
- [x] Update tests to use new metadata store + storage connectors 

Remaining todos: 
* Adapt `record.text` to be `record.content`, which can be a List of strings. 
* Update all instances of `create_agent_from_preset` to use correct format
* Remove all references of `AgentConfig` and replace them with `AgentState`, which can be looked up from the metadata store 

---

**How to test**

- Regression test (unit tests should cover)
- Make sure that persistence manager is persisting (manual DB inspection and/or save+load)

**Have you tested this PR?**

WIP
